File: ruby/series.rb
class Series
  attr_reader :digits

  def initialize(digits)
    @digits = digits
  end

  def slices(size)
    raise ArgumentError, "Invalid slice size" if size > digits.length
    sliced_digits = []
    digits_to_slice = digits.split("")
    until digits_to_slice.length < size
      sliced_digits<<digits_to_slice.first(size).join
      digits_to_slice.shift
    end
    sliced_digits.flatten
  end
end

File: js/series.js
Series = function(digits){
  this.digits=separateDigits(digits);
};

function separateDigits(digits){
  var separated = digits.split("")
  var transformed = separated.map(function(num){
    return parseInt(num, 10);
  });
  return transformed;
};

Series.prototype.slices = function(size, output=[]){
  if (size>this.digits.length){
    throw ('Slice size is too big.');
  };
  for (i=0; i+size<=this.digits.length; i++){
    output.push(this.digits.slice(i, i+size));
  };
  return output
};

module.exports = Series;
